nomad/structs/services.go

```diff
@@ -1123,13 +1136,51 @@ func (s *Service) Equal(o *Service) bool {
 		return false
 	}
 
+	if !s.Weights.Equal(o.Weights) {
+		return false
+	}
+
 	return true
 }
 
 func (s *Service) IsConsul() bool {
 	return s.Provider == ServiceProviderConsul || s.Provider == ""
 }
 
+// ConsulWeights represents a Consul weights for a service block.
+type ConsulWeights struct {
```

Comment:
If we call this struct `ConsulWeights`, we're stuck with it if we later decide to support DNS and weights in Nomad service discovery. Let's call this `ServiceWeights` instead (that'll ripple through the rest of the PR as well).